### PR TITLE
[android] add android makefile and check

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: Android.mk [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    $(LOCAL_PATH)/src \
+    $(LOCAL_PATH)/third_party/wpantund \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/ipc-dbus \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/util \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/wpantund \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/wpanctl \
+    $(LOCAL_PATH)/third_party/wpantund/repo/third_party/assert-macros \
+    $(LOCAL_PATH)/third_party/wpantund/repo/third_party/openthread/src/ncp \
+	$(NULL)
```
Comment (jwhui) [inline Android.mk]: nit: alignment

File: Android.mk [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+LOCAL_SRC_FILES := \
+    third_party/wpantund/repo/src/wpanctl/wpanctl-utils.c \
+    src/agent/agent_instance.cpp \
+    src/agent/border_agent.cpp \
+    src/agent/mdns_mdnssd.cpp \
+    src/agent/main.cpp \
+    src/agent/ncp_wpantund.cpp \
+	src/common/event_emitter.cpp \
```
Comment (jwhui) [inline Android.mk]: nit: alignment

File: Android.mk [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/util \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/wpantund \
+    $(LOCAL_PATH)/third_party/wpantund/repo/src/wpanctl \
+    $(LOCAL_PATH)/third_party/wpantund/repo/third_party/assert-macros \
+    $(LOCAL_PATH)/third_party/wpantund/repo/third_party/openthread/src/ncp \
+	$(NULL)
+
+LOCAL_CFLAGS := \
+    -DPACKAGE_VERSION=\"0.01.00\" \
+	$(NULL)
```
Comment (jwhui) [inline Android.mk]: nit: alignment

**Comment by bukepo** [comment]
I have removed the generated files from the script.

**Pull request merged**